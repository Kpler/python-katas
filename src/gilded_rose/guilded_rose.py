class GildedRose(object):

    def __init__(self, items):
        self.items = items

    def update_quality(self):
        for item in self.items:
            if item.name != "Aged Brie" and item.name != "Backstage passes to a TAFKAL80ETC concert":
                if item.quality > 0:
                    if item.name != "Sulfuras, Hand of Ragnaros":
                        item.quality = item.quality - 1
            else:
                if item.quality < 50:
                    if item.name == "Backstage passes to a TAFKAL80ETC concert":
                        if item.sell_in < 6:
                            item.quality = item.quality + 3
                        elif item.sell_in < 11:
                            item.quality = item.quality + 2
                        else:
                            item.quality = item.quality + 1
                    else:
                        item.quality = item.quality + 1
            if item.name != "Sulfuras, Hand of Ragnaros":
                item.sell_in = item.sell_in - 1
            if item.sell_in < 0:
                if item.name != "Aged Brie":
                    if item.name != "Backstage passes to a TAFKAL80ETC concert":
                        if item.quality > 0:
                            if item.name != "Sulfuras, Hand of Ragnaros":
                                item.quality = item.quality - 1
                    else:
                        item.quality = item.quality - item.quality
                else:
                    if item.quality < 50:
                        item.quality = item.quality + 1


class Item:
    def __init__(self, name, sell_in, quality):
        self.name = name
        self.sell_in = sell_in
        self.quality = quality

    def __repr__(self):
        return "%s, %s, %s\n" % (self.name, self.sell_in, self.quality)
