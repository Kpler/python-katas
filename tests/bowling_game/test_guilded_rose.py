from approvaltests.combination_approvals import verify_all_combinations

from src.gilded_rose.guilded_rose import Item, GildedRose


def test_gilded_rose_approval_tests():
    verify_all_combinations(
        execute_update_quality,
        [['Aged Brie'],
        [1],
        [1]]
    )


def execute_update_quality(name: str, sell_in: int, quality: int) -> Item:
    items_under_test = [
        Item(name=name, sell_in=sell_in, quality=quality),
    ]
    gilded_rose = GildedRose(items_under_test)
    gilded_rose.update_quality()
    return gilded_rose.items[0]
