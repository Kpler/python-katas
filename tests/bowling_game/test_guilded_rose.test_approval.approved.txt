[Aged Brie, 0, 2]
